source/requests/streams.d

```diff
@@ -1131,34 +1131,28 @@ public abstract class SocketStream : NetworkStream {
     in {assert(isConnected);}
     do {
         auto rc = s.send(buff);
-        if (rc < 0) {
-            close();
-            throw new NetworkException("sending data: %s".format(to!string(strerror(errno))));
+        if (rc <= 0) {
+            rc = 0;
+            const code = openssl.ERR_get_error();
+            if(code != 0) {
+                const msg = openssl.ERR_reason_error_string(code);
+                close();
+                throw new NetworkException("sending data: "~msg.to!string);
+            }
         }
         return rc;
     }
 
     ptrdiff_t receive(void[] buff) {
         while (true) {
             auto r = s.receive(buff);
-            if (r < 0) {
-                auto e = errno;
-                version(Windows) {
+            if (r <= 0) {
+                r = 0;
+                const code = openssl.ERR_get_error();
+                if(code != 0) {
+                    const msg = openssl.ERR_reason_error_string(code);
                     close();
-                    if ( e == 0 ) {
-                        throw new TimeoutException("Timeout receiving data");
-                    }
-                    throw new NetworkException("Unexpected error %s while receiving data".format(to!string(strerror(errno))));
-                }
-                version(Posix) {
-                    if ( e == EINTR ) {
-                        continue;
```

Comment:
Is it not relevant anymore?
See for example https://www.reddit.com/r/C_Programming/comments/o58l5q/should_i_retry_calls_that_fail_and_errno_equals/

Reply:
I don't know. I think SSL on its level catches these things

Reply:
Comment on Reddit:

_Although I wouldn’t expect EINTR to continue to be thrown over and over, it might under a meltdown._

Reply:
I recal there was reasons for this EINTR case  handling. Can we leave it as is somehow?

Reply:
EINTR check added

OpenSSL have internal `get_last_socket_error_is_eintr()` and they actively deal with EINTR, so I don't know this is necessary or not